File: gommo-backend/src/main/java/br/com/gommo/modules/attendance/mapper/AttendanceRecordMapper.java
package br.com.gommo.modules.attendance.mapper;
import br.com.gommo.modules.attendance.dto.*; import br.com.gommo.modules.attendance.entity.AttendanceRecord; import org.springframework.stereotype.Component;
@Component public class AttendanceRecordMapper {
    public AttendanceRecord toEntity(AttendanceRecordRequestDto dto) {
        return AttendanceRecord.builder().collaboratorId(dto.getCollaboratorId()).workDate(dto.getWorkDate())
            .clockIn(dto.getClockIn()).clockOut(dto.getClockOut()).breakMinutes(dto.getBreakMinutes()).notes(dto.getNotes()).build();
    }
    public void updateEntity(AttendanceRecord entity, AttendanceRecordRequestDto dto) {
        entity.setCollaboratorId(dto.getCollaboratorId()); entity.setWorkDate(dto.getWorkDate());
        entity.setClockIn(dto.getClockIn()); entity.setClockOut(dto.getClockOut());
        entity.setBreakMinutes(dto.getBreakMinutes()); entity.setNotes(dto.getNotes());
    }
    public AttendanceRecordResponseDto toResponse(AttendanceRecord entity) {
        return AttendanceRecordResponseDto.builder().id(entity.getId()).status(entity.getStatus())
            .collaboratorId(entity.getCollaboratorId()).workDate(entity.getWorkDate()).clockIn(entity.getClockIn())
            .clockOut(entity.getClockOut()).breakMinutes(entity.getBreakMinutes()).notes(entity.getNotes())
            .createdAt(entity.getCreatedAt()).updatedAt(entity.getUpdatedAt()).build();
    }
}

File: gommo-frontend/src/modules/person/attendance/lib/attendance-record.mapper.ts
import type { AttendanceRecord, AttendanceRecordCreateDto } from "@/modules/person/attendance/dto/attendance-record.dto";

export function attendancerecordToFormDto(entity: AttendanceRecord): AttendanceRecordCreateDto {
    return {
        collaboratorId: entity.collaboratorId ?? "",
        workDate: entity.workDate?.slice(0, 10) ?? "",
        clockIn: entity.clockIn ?? "",
        clockOut: entity.clockOut ?? "",
    };
}

export const emptyAttendanceRecordForm = (): AttendanceRecordCreateDto => ({
    collaboratorId: "",
    workDate: "",
    clockIn: "",
    clockOut: "",
});
